src/main/java/com/github/rjeschke/txtmark/Line.java

```diff
@@ -296,6 +296,12 @@ public LineType getLineType(final Configuration configuration)
             return LineType.BQUOTE;
         }
 
+        if(this.leading == 0 && this.value.matches("(.+\\|)+.+") && this.next != null && this.next.value != null && this.next.value.matches("(-+\\|)*-*")){
```

Comment:
Table in Markdown may be better to be like 

|title1|title2|
|--|--|
|value1|value2|

It's easy to parse and more pretty

Reply:
For first land and second line should be:
| Tables     |      Are            |  Cool  |
|----------|:-------------:|------:|

if (this.leading == 0 && this.value.matches("^\\|(.+\\|)+.+\\|$")){
            // First Line Looks Like:  | Tables   |      Are      |  Cool |
            if (this.next != null && this.next.value != null && this.next.value.matches("^\\|(-|:|\\|)+\\|$")) {
                // Second Line Looks Like: |----------|:-------------:|------:|
                return LineType.TABLE;
            }
        }

Please refer: **https://www.tablesgenerator.com/markdown_tables** to generator markdown table

Reply:
Please refer: https://www.tablesgenerator.com/markdown_tables to generator markdown table.
The first line and second line should be:
````
| Tables   |      Are      |  Cool |
|----------|:-------------:|------:|
````

```
if (this.leading == 0 && this.value.matches("^\\|(.+\\|)+.+\\|$")){
      // First Line Looks Like:  | Tables   |      Are      |  Cool |
      if (this.next != null && this.next.value != null && this.next.value.matches("^\\|(-|:|\\|)+\\|$")) {
            // Second Line Looks Like: |----------|:-------------:|------:|
            return LineType.TABLE;
      }
}
````